### DISCUSSION TITLE
Add Pressure sensor to "Analog Group"

### DISCUSSION THREAD

**Discussion by trlafleur** [arendst/Tasmota#13310] [Ideas]
Another sensor type that could easily be added to the “Analog Group” is a pressure sensor. 
They are available from many sources, from China, Amazon, eBay, and other commercial retailers. They can be used for many applications from monitoring water pressure to oil or gas pressure.

I use them for measuring water pressure from the local utility, on pool pumps, and for well pump pressure.

One can get them with a full-scale reading of a few PSI to well over 1000 PSI  (or Bar..)
As an example:

"Pressure Range: 0-100 psi Material : Stainless steel Output: 0. 5V - 4. 5V linear voltage output. 0 psi outputs 0. 5V, 50 psi outputs 2. 5V, 100 psi outputs 4. 5V Working Temperature: -40-+125C;. Works for oil, fuel, water, or air pressure. Can be used in oil tank, gas tank, etc. Accuracy: 1%FS; Thread: 1/8"-27 NPT"

They all have a standard 3 wire interface, +5v, Sensor out, and Gnd. (there are some 12V versions available, but they are not common)

Adapting to the “Analog Group” would require,

1)	Scaling resistors ratios for the ATD input for ESP8266 and ESP32
2)	Adding a new variable to select max PSI
3)	Scaling of voltage from 0.5v to 4.5v to give or 0 to 100PSI. (or 0 to 100%)
4)	Adding a calibration offset variable.
5)	Selecting units for display from PSI, BAR or ??
Some of the above functions are already available in the Analog sensor code.

~~~
https://www.amazon.com/AUTEX-Stainless-Transducer-Transmitter-compatible/dp/B00NIK900K
https://www.amazon.com/IIL-Pressure-transducer-Sender-Water/dp/B07GWJ1PHD
https://www.amazon.com/100psi-Precision-Stainless-Pressure-Transducer/dp/B01FTY1DHK

HP4440 Pressure Sensor Connector 

~~~

**Comment by barbudor** [comment]
Hi Tom,
ADC Mode 6 already offer full scaling/mapping : https://tasmota.github.io/docs/Commands/#adcparam
What would you suggest the SENSOR message to look like for point 5) ?
Today it's `{...,"ANALOG":{"Range":1}}`

**Reply by sfromis** [reply]
I suppose that the docs could be amended with examples of AdcParam usage for such pressure sensors, including the small difference between unit psi or bar. Of course, the unit would not be included in the JSON payload or the web page display, with the latter generally being seen as "not for daily usage".

**Comment by trlafleur** [comment]
I think it should report "Pressure" and the scale units...PSI, Bar ect...

~~ _/) ~~~~ _/) ~~~~ _/) ~~~~ _/) ~~

Tom Lafleur

**Reply by barbudor** [reply]
Let's imagine an extension of `AdcParam 6`with an optional name argument:
`AdcParam 6, adcMin, adcMax, rangeMin, rangeMax [, name]`
`AdcParam 6, 50, 1024, 0, 100, Pressure`
could return: `{...,"ANALOG":{"Pressure":1}}`

But the unit may be a problem as Tasmota reports already temperature and pressure units for sensors like in below:
```
{
  "Time": "2021-10-09T18:10:10+02:00",
  "Charging": "ON",
  "Charged": "OFF",
  "ANALOG": {
    "Range": 4236
  },
  "BME280": {
    "Temperature": 17.1,
    "Humidity": 100.0,
    "DewPoint": 17.1,
    "Pressure": 1006.8,
    "SeaPressure": 1028.1
  },
  "TSL2561": {
    "Illuminance": 114.2,
    "IR": 7377,
    "Broadband": 14993
  },
  "PressureUnit": "hPa",
  "TempUnit": "C"
}
```

Would giving the unit in the name enough ? `{...,"ANALOG":{"Pressure (PSI)":1}}`

**Comment by trlafleur** [comment]
if Pressure is reported in the analog space, I don't this it would be an
issue....

"ANALOG": {
    "Range": 4236,

        "Pressure": 234
  },


~~ _/) ~~~~ _/) ~~~~ _/) ~~~~ _/) ~~

Tom Lafleur

**Reply by barbudor** [reply]
What would Range mean here ?
If you apply a range, it is to scale/map the ADC value to that range
So only 1 value, the mapped one, would be needed.
